consensus/src/metrics_safety_rules.rs

```diff
@@ -125,3 +144,137 @@ impl TSafetyRules for MetricsSafetyRules {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{metrics_safety_rules::MetricsSafetyRules, test_utils::EmptyStorage};
+    use claim::{assert_matches, assert_ok};
+    use consensus_types::{
+        block_data::BlockData,
+        safety_data::SafetyData,
+        timeout::Timeout,
+        timeout_2chain::{TwoChainTimeout, TwoChainTimeoutCertificate},
+        vote::Vote,
+        vote_proposal::MaybeSignedVoteProposal,
+    };
+    use diem_crypto::{ed25519::Ed25519Signature, HashValue};
+    use diem_types::{
+        block_info::BlockInfo,
+        epoch_change::EpochChangeProof,
+        ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+        waypoint::Waypoint,
+    };
+    use safety_rules::{ConsensusState, Error, TSafetyRules};
+
+    pub struct MockSafetyRules {
+        // number of initialize() calls
+        init_calls: i32,
+
+        // max initialize() calls to complete perform_initialize()
+        max_init_calls: i32,
+
+        // last initialize() returns Ok() or any error != WaypointOutOfDate
+        last_init_result: Result<(), Error>,
+    }
+
+    impl MockSafetyRules {
+        pub fn new(
+            init_calls: i32,
+            max_init_calls: i32,
+            last_init_result: Result<(), Error>,
+        ) -> Self {
+            Self {
+                init_calls,
+                max_init_calls,
+                last_init_result,
+            }
+        }
+    }
+
+    impl TSafetyRules for MockSafetyRules {
+        fn consensus_state(&mut self) -> Result<ConsensusState, Error> {
+            let safety_data = SafetyData::new(0, 0, 0, 0, None);
+            let waypoint = Waypoint::new_any(&LedgerInfo::new(
+                BlockInfo::new(0, 0, HashValue::zero(), HashValue::zero(), 0, 0, None),
+                HashValue::zero(),
+            ));
+            Ok(ConsensusState::new(safety_data, waypoint, true))
```

Comment:
I think we can just use ConsensusState::default()

Reply:
This (`ConsensusState::default()`) is the true lifesaver! Thanks @zekun000 !!